src/templates/rule.js

```diff
@@ -194,9 +194,23 @@ const Rule = ({ data, location }) => {
               <ol>
                 {rule.frontmatter.related
                   ? rule.frontmatter.related.map((relatedRuleUri) => {
-                      const relatedRule = data.relatedRules.nodes.find(
-                        (r) => r.frontmatter.uri === relatedRuleUri
+                      const allRelatedRules = data.relatedRules.nodes.concat(
+                        data.relatedRulesFromRedirects.nodes
                       );
+                      const relatedRule = allRelatedRules.find((r) => {
+                        if (r.frontmatter.uri === relatedRuleUri) {
+                          return r;
+                        } else {
+                          var relatedRuleFromRedirect;
+                          r.frontmatter.redirects &&
+                            r.frontmatter.redirects.find((redirect) => {
+                              if (redirect === relatedRuleUri) {
+                                relatedRuleFromRedirect = r;
+                              }
+                            });
+                          return relatedRuleFromRedirect;
+                        }
+                      });
```

Comment:
This passage is confusing, it is difficult to figure out what is going on. I think you should break this out into functions and I also think you should do the check for the exact rule separately from the rule URIs. Call me if you need further clarification